easy_phi/app.py

```diff
@@ -96,26 +100,33 @@ def get(self):
 
 class SelectModuleHandler(ModuleHandler):
     """Select the module, identified by ID passed as a URL parameter.
-    get, head: get status
-    put:    set lock
-    delete: remove lock
-    post:   change status
+
+    When user selects module in web interface, all other users should be
+    be able to see this module is used by someone else.
+
     """
     allow_broadcast = False
 
+    @api_auth
     def post(self):
         """ Set user lock on module to indicate it is used by someone """
-        session = self.get_argument("session")
-        # Check the status of selected module (must be unlocked) [TBD]
-
-        # Lock the module [TBD]
-
-        # Redirect user to a constructed Module specific web-page [TBD]
-
+        used_by = getattr(self.module, 'used_by', None)
+        if used_by is not None:
+            self.set_status(400)
+            return {'error': 'Module is used by {0}. If you '.format(used_by) +
+                             'need this module, you might force unlock it '
+                             'by issuing DELETE request first.'}
```

Comment:
As far as I remember it can be unlocked by anyone. Actually, it is a requirements issue - at least, this decision was not documented.
I propose to leave it for now as security backend and whole concept of admin/non-admin users is not clear, and we don't have clear requirements. Meanwhile, we'll need to clarify this point on next meeting